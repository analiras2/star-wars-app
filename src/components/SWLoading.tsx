import React, { PureComponent } from 'react';
import { ActivityIndicator } from 'react-native';
import colors from '../res/colors';
import SWText from './SWText';
import SWView from './SWView';

interface Props {
  title: string;
}
export default class SWLoading extends PureComponent<Props> {
  public render() {
    return (
      <SWView>
        <ActivityIndicator color={colors.accent} size={64} />
        <SWText
          title={this.props.title.toUpperCase()}
          textAlign="center"
          color={colors.accent}
          small={true}
        />
      </SWView>
    );
  }
}
