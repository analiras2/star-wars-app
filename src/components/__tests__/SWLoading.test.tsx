import { shallow } from 'enzyme';
import React from 'react';
import { render } from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import SWLoading from '../SWLoading';

describe('SWLoading', () => {
  const props = { title: 'Loading' };
  const wrapper = shallow(<SWLoading {...props} />);
  const { getByText } = render(<SWLoading {...props} />);

  describe('validating SWLoading texts', () => {
    it('should render Loading', () => {
      expect(getByText(/Loading/i)).toBeDefined();
    });
  });

  describe('rendering Components', () => {
    it('should render a <SWView />', () => {
      expect(wrapper.find('SWView')).toHaveLength(1);
    });

    it('should render a <SWText />', () => {
      expect(wrapper.find('SWText')).toHaveLength(1);
    });
  });

  it('renders SWLoading correctly', () => {
    const tree = renderer.create(<SWLoading {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
