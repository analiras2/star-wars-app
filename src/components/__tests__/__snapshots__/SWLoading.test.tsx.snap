// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SWLoading renders SWLoading correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "flex": 0,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#FFE81F"
    hidesWhenStopped={true}
    size={64}
  />
  <Text
    style={
      Object {
        "alignSelf": "auto",
        "color": "#FFE81F",
        "flex": 0,
        "fontFamily": "Exo-Regular",
        "fontSize": 12,
        "letterSpacing": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "textAlign": "center",
      }
    }
  >
    LOADING
  </Text>
</View>
`;
